flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_MainDisplay.js

```diff
@@ -1378,135 +1213,160 @@ class A320_Neo_CDU_MainDisplay extends FMCMainDisplay {
+        mainframe.appendChild(this.mcduRight.root);
+
+        this.mcduLeft.Init();
+        this.mcduRight.Init();
+
+        this.aocAirportList = new CDUAocAirportList; // TODO kill this ugliness
+
+        // If the consent is not set, show telex page
+        const onlineFeaturesStatus = NXDataStore.get("CONFIG_ONLINE_FEATURES_STATUS", "UNKNOWN");
+        if (onlineFeaturesStatus === "UNKNOWN") {
+            CDU_OPTIONS_TELEX.ShowPage(this.fmc, this);
         }
-        return messages[currentMessageIndex];
     }
 
-    getMessageIndex(id) {
-        return this.messages.findIndex(m => m["id"].toString() === id.toString());
+    onUpdate(_deltaTime) {
+        super.onUpdate(_deltaTime);
+
+        this.mcduLeft.onUpdate(_deltaTime);
+        this.mcduRight.onUpdate(_deltaTime);
     }
 
-    addMessage(message) {
-        this.messages.unshift(message);
-        const cMsgCnt = SimVar.GetSimVarValue("L:A32NX_COMPANY_MSG_COUNT", "Number");
-        SimVar.SetSimVarValue("L:A32NX_COMPANY_MSG_COUNT", "Number", cMsgCnt + 1);
-        if (this.refreshPageCallback) {
-            this.refreshPageCallback();
-        }
+    initMcduVariables() {
+        this.mcduLeft.clearMessageQueue();
+        this.mcduRight.clearMessageQueue();
+        this.aocAirportList.init();
     }
 
-    deleteMessage(id) {
-        if (!this.messages[id]["opened"]) {
-            const cMsgCnt = SimVar.GetSimVarValue("L:A32NX_COMPANY_MSG_COUNT", "Number");
-            SimVar.SetSimVarValue("L:A32NX_COMPANY_MSG_COUNT", "Number", cMsgCnt <= 1 ? 0 : cMsgCnt - 1);
-        }
-        this.messages.splice(id, 1);
+    /**
+     * Updates both MCDU displays
+     */
+    requestUpdate() {
+        this.requestLeftUpdate();
+        this.requestRightUpdate();
     }
 
-    // OUTGOING/SENT AOC MESSAGES
-    getSentMessages() {
-        return this.sentMessages;
+    requestLeftUpdate() {
+        this.mcduLeft.requestOnsideUpdate();
     }
 
-    getSentMessage(id, type) {
-        const messages = this.sentMessages;
-        const currentMessageIndex = messages.findIndex(m => m["id"].toString() === id.toString());
-        if (type === 'previous') {
-            if (messages[currentMessageIndex - 1]) {
-                return messages[currentMessageIndex - 1];
-            }
-            return null;
-        } else if (type === 'next') {
-            if (messages[currentMessageIndex + 1]) {
-                return messages[currentMessageIndex + 1];
-            }
-            return null;
+    requestRightUpdate() {
+        this.mcduRight.requestOnsideUpdate();
+    }
+
+    onEvent(_event) {
+        super.onEvent(_event);
+        console.log(_event);
+        if (_event.indexOf("1_BTN_") !== -1) {
+            const input = _event.replace("1_BTN_", "");
+            this.mcduLeft.onEvent(input);
+        } else if (_event.indexOf("2_BTN_") !== -1) {
+            const input = _event.replace("2_BTN_", "");
+            this.mcduRight.onEvent(input);
```

Comment:
Just typing some code to spawn more ideas. Don't have to do anything with this 😀.

```suggestion
    onEvent(_event) {
        super.onEvent(_event);
        console.log(_event);
        this.mcdus.forEach((mcdu, index) => {
            if (_event.indexOf(`${index + 1}_BTN_`) !== -1) {
                mcdu.onEvent(_event.replace(`${index + 1}_BTN_`, ''));
            }
        });
```